Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -531,22 +529,17 @@ extension ChannelDTO {
 
 private extension ChannelDTO {
     /// Updates the `oldestMessageAt` of the channel. It should only updates if the current `messages: [Message]`
-    /// is older than the current `ChannelDTO.oldestMessageAt`.
-    /// If the response is not paginated, we need to reset `oldestMessageAt` to the oldest message in the payload.
-    func updateOldestMessageAt(payload: ChannelPayload, isPaginatedPayload: Bool) {
-        guard let payloadOldestMessageAt = payload.messages.map(\.createdAt).min() else { return }
-
-        guard isPaginatedPayload else {
-            oldestMessageAt = payloadOldestMessageAt.bridgeDate
-            return
-        }
-
-        let isOlderThanCurrentOldestMessage = payloadOldestMessageAt < (oldestMessageAt?.bridgeDate ?? Date.distantFuture)
-        if isOlderThanCurrentOldestMessage {
-            oldestMessageAt = payloadOldestMessageAt.bridgeDate
+    /// is older than the current `ChannelDTO.oldestMessageAt`, unless the current `ChannelDTO.oldestMessageAt`
+    /// is the default one, which is by default a very old date, so are sure the first messages are always fetched.
+    func updateOldestMessageAt(payload: ChannelPayload) {
+        if let payloadOldestMessageAt = payload.messages.map(\.createdAt).min() {
+            let isOlderThanCurrentOldestMessage = payloadOldestMessageAt < (oldestMessageAt?.bridgeDate ?? Date.distantFuture)
+            if isOlderThanCurrentOldestMessage {
+                oldestMessageAt = payloadOldestMessageAt.bridgeDate
+            }
```

Comment:
Note to self: Use a guard here and explain why we need `isOlderThanCurrentOldestMessage`